.github/workflows/python-cli-build-and-test.yml

```diff
@@ -0,0 +1,84 @@
+name: Build and test Python CLI
+
+on:
```

Comment:
hmm do we know if this workflow fails does that affect any of the other workflows?  I assume if this fails the tag-publish workflow would still do its thing?

Reply:
yeah this is independent of the other workflows